### ISSUE TITLE
Requesting re-authentication from oidc clients

### ISSUE THREAD

**Issue by Jeidnx** [kanidm/kanidm#3206]
### Is your feature request related to a problem? Please describe

Currently there is no way to force a user to be re-authenticated from an oidc client, see also #2988.

### Describe the solution you'd like

Full support for the `max_age` parameter, including returning the `auth_time` claim.
The differences between the `prompt` and `max_age` methods are (afaict) that with `prompt` a client can't be sure if the request was honored. When `max_age` is supported, the token response *must* contain an `auth_time` claim, which the client can use to verify that the user was indeed re-authenticated.

### Describe alternatives you've considered

#2988 may be easier to implement, but imo having the ability to verify if the user was re-authenticated is a useful feature.

### Additional context

Kind of related to #3185, and as already mentioned there i would be willing to take a shot at implementing this, although after looking at the code more this may be above my head.

**Comment by Firstyear** [comment]
Yeah, this one may be a bit tricky. We have had this request made before, and we just haven't yet been able to prioritise it. 

The hardest part of this would be we need a way to start the re-auth / login flow, but if the user cancels it, it doesn't cancel the actual live session they are part of, and that the re-auth process shouldn't trigger a need for elevated rights. 

So I'd say we need some extra steps - probably similar to how we handle the consent flow. We have to choose if consent is before or after the re-auth. We also then would need to change re-auth to not alter the session priv state.

**Comment by BrainWart** [comment]
There's documentation by Auth0¹ which seems to suggest the same or similar as the OP.

1 - https://auth0.com/docs/authenticate/login/max-age-reauthentication

**Comment by yaleman** [comment]
I agree we need to implement it, but as @Firstyear explained, we need to work out what that looks like as far as user workflows/stories in Kanidm 😄

**Comment by Firstyear** [comment]
This is now supported via oidc prompt=login.
